### PR TITLE
Tickets/dm 7380

### PR DESCRIPTION
**Pull request by jgates108**

### CODE REVIEW

File: core/modules/qdisp/QueryResource.cc [hunk 1]
```diff
@@ -64,14 +64,15 @@ char const* QueryResource::_getRname(std::string const& rName) {
...
+    LOGS(_log, LOG_LVL_DEBUG, _jobIdStr << "~QueryResource()");
     free(_rNameHolder);
+    _rNameHolder = nullptr;
 }
 
 /// May not throw exceptions because the calling code comes from
 /// xrootd land and will not catch any exceptions.
 void QueryResource::ProvisionDone(XrdSsiSession* s) {
-    LOGS_DEBUG(_jobIdStr << "QueryResource::ProvisionDone ");
+    LOGS_DEBUG(_jobIdStr << " QueryResource::ProvisionDone ");
```
Comment (andy-slac) [inline core/modules/qdisp/QueryResource.cc]: One more space left :smirk_cat:

File: core/modules/wpublish/QueriesAndChunks.cc [hunk 2]
```diff
@@ -229,6 +231,7 @@ QueryStatistics::Ptr QueriesAndChunks::getStats(QueryId const& qId) const {
 void QueriesAndChunks::examineAll() {
     // Need to know how long it takes to complete tasks on each table
     // in each chunk, and their percentage total of the whole.
+    std::cout << "&&& QueriesAndChunks::examineAll a\n";
```
Comment (andy-slac) [inline core/modules/wpublish/QueriesAndChunks.cc]: Should it use logger instead of `cout`?

File: core/modules/wsched/testSchedulers.cc [hunk 3]
```diff
@@ -817,13 +817,23 @@ BOOST_AUTO_TEST_CASE(BlendScheduleQueryBootTaskTest) {
     }
     running = false;
     // f.queries should now have a baseline for chunk 27.
+    LOGS(_log, LOG_LVL_DEBUG, "Chunks after fastFunc " << *f.queries);
+    std::cout << "Chunks after fastFunc " << *f.queries << "\n"; // &&&
```
Comment (andy-slac) [inline core/modules/wsched/testSchedulers.cc]: Why do you do both logger and `cout`?

File: core/modules/wpublish/QueriesAndChunks.cc (hunk 2 above)
Comment (jgates108) [inline core/modules/wpublish/QueriesAndChunks.cc]: I removed everything with "&&&" and did not push it to master after I committed. Basically, I just removed debugging instrumentation. The cout was a temporary workaround for the logger not working on a unit test. I just pushed the correct version to master.

File: core/modules/qdisp/Executive.h [hunk 4]
```diff
@@ -75,7 +75,7 @@ class Executive {
     /// Construct an Executive.
     /// If c->serviceUrl == Config::getMockStr(), then use XrdSsiServiceMock
     /// instead of a real XrdSsiService
-    Executive(Config::Ptr c, std::shared_ptr<MessageStore> ms);
+    static Executive::Ptr newExecutive(Config::Ptr const& c, std::shared_ptr<MessageStore> const& ms);
```
Comment (jgates108) [inline core/modules/qdisp/Executive.h]: For enabled_shared_from_this to work properly, the object must be created as a shared pointer. There's a problem if someone does
<code>
    Executive* exec = new Executive(...);
<code>
So making a private constructor and the factory function provides a reasonable amount of protection. Unfortunately, std::make_shared really doesn't like private constructors, the factory function just uses new.

File: core/modules/qdisp/QueryRequest.cc [hunk 5]
```diff
@@ -76,15 +76,16 @@ QueryRequest::~QueryRequest() {
...
         LOGS(_log, LOG_LVL_DEBUG, _jobIdStr << " QueryRequest::GetRequest called after job finished (cancelled?)");
         requestLength = 0;
         return const_cast<char*>("");
     }
-    requestLength = _jobQuery->getDescription().payload().size();
+    requestLength = jq->getDescription().payload().size();
     LOGS(_log, LOG_LVL_DEBUG, _jobIdStr << " Requesting, payload size: " << requestLength);
     // Andy promises that his code won't corrupt it.
-    return const_cast<char*>(_jobQuery->getDescription().payload().data());
+    return const_cast<char*>(jq->getDescription().payload().data());
```
Comment (jgates108) [inline core/modules/qdisp/QueryRequest.cc]: QueryRequest::GetRequest() is an overloaded function from xrootd. We have to conform to the API.

File: core/modules/qdisp/QueryResource.cc (hunk 1 above)
Comment (jgates108) [inline core/modules/qdisp/QueryResource.cc]: I think Andy was talking about the space at the end.

**Pull request merged**